usr/lib/common/pkey_utils.c

```diff
@@ -1200,7 +1200,7 @@ CK_RV pkey_ec_sign(STDLL_TokData_t *tokdata, SESSION *session,
             hash_len = 32;
         off = 32 - hash_len;
         memcpy(param.P256.hash + off, hash, hash_len);
-        memcpy(param.P256.priv, pkey_attr->pValue, 32);
+        memcpy(param.P256.priv, pkey_attr->pValue, MIN(pkey_attr->ulValueLen - 32, 32));
         memcpy(param.P256.vp, (char *)pkey_attr->pValue + 32, 32);
```

Comment:
Use `MIN(..., sizeof(param.P256.vp))` here as well.
Also, did you check anywhere that `pkey_attr->ulValueLen` is at least 32+32 bytes in size? Otherwise you access memory after the actual allocation.